cloud/pkg/common/messagelayer/util.go

```diff
@@ -131,8 +131,8 @@ func BuildResourceForDevice(nodeID, resourceType, resourceID string) (resource s
 // GetDeviceID returns the ID of the device
 func GetDeviceID(resource string) (string, error) {
 	res := strings.Split(resource, "/")
-	if len(res) >= ResourceDeviceIDIndex+1 && res[ResourceDeviceIndex] == ResourceDevice {
-		return res[ResourceDeviceIDIndex], nil
+	if len(res) >= ResourceDeviceIDIndex+2 && res[ResourceDeviceIndex] == ResourceDevice {
```

Comment:
You can define some constants to replace magic numbers or add some comments.

Reply:
ok，I have made it.